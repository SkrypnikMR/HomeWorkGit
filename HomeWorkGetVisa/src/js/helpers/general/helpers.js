import * as validate from "../validators/validators.js";
import { notifKill } from "../listeners/listeners.js";
import * as drowers from "../drowers/drowers.js";

export const click = (buttons) => {
  for (let key in buttons) {
    buttons[key].click();
  }
};

export const validator = (candidate) => {
  const vName = validate.validateName(candidate.name);
  const vBalance = validate.validateNumber(candidate.balance);
  const vAgelikeNumber = validate.validateNumber(candidate.age);
  const VAgeLength = validate.validateLength(candidate.age, 3);
  const vEnglish = validate.validateEnglish(candidate.english);
  if (vName || !vBalance || !vAgelikeNumber || VAgeLength || !vEnglish) {
    return false;
  }
  return true;
};

export const cleaner = (nodes) => {
  for (let key in nodes) {
    if (nodes[key].checked === true) {
      nodes[key].checked = false;
      continue;
    }
    nodes[key].value = "";
  }
};

export const createNotif = (str) => {
  // create notif Div
  const notif = document.createElement("div");
  const notifTextDiv = document.createElement("div");
  const notifButtonDiv = document.createElement("div");
  const notifDeleteBtn = document.createElement("button");
  const notifMessage = document.createElement("p");

  // add classes
  notif.classList.add("notification");
  notifButtonDiv.classList.add("notification__button");
  notifTextDiv.classList.add("notification__message");
  notifDeleteBtn.classList.add("button__item");
  notifDeleteBtn.classList.add("button");
  notifMessage.classList.add("message__item");

  //add textContent
  notifDeleteBtn.textContent = "X";
  notifMessage.textContent = str;

  //append chields
  notifTextDiv.appendChild(notifMessage);
  notifButtonDiv.appendChild(notifDeleteBtn);
  notif.appendChild(notifButtonDiv);
  notif.appendChild(notifTextDiv);
  document.body.appendChild(notif);

  // add event Listener
  notifDeleteBtn.addEventListener("click", notifKill.bind(null, notif));

  // autoDeath after 3s
  setTimeout(notifKill.bind(null, notif), 2000);
};

export const mutateObject = (obj) => {
  obj.balance = Number(obj.balance);
  obj.age = Number(obj.age);
  return obj;
};

export const drowInit = (candidates, canvas, context, distance_Between) => {
  drowers.drowBackground(canvas, context);
  for (let i = 0; i < candidates.length; i++) {
    const distance = distance_Between * (i + 1) + 2;
    const x = 0;
    const k = (2 * Math.PI) / 100;
    drowers.drowText(context, candidates[i].name, distance, 92, "white", 12);
    drowers.drowText(context, "balance", distance, 292, "white", 12);
    drowers.drowText(context, "documents", distance, 492, "white", 12);
    drowers.drowText(context, "age", distance, 692, "white", 12);
    drowers.drowText(context, "english", distance, 892, "white", 12);
    drowers.drowCircle(context, canvas, x, k, 100, distance, 100, 0, "green");
    candidates[i].distance = distance;
    candidates[i].balancePoint = 292;
    candidates[i].documentsPoint = 492;
    candidates[i].agePoint = 692;
    candidates[i].englishPoint = 892;
  }
};

export const sessionStorageSet = (key, value) =>
  sessionStorage.setItem(key, value);
