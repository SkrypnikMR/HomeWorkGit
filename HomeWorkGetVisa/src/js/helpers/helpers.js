import * as validate from "./validators/validators.js";
import { notifKill } from "./listeners.js";

export const click = (buttons) => {
  for (let key in buttons) {
    buttons[key].click();
  }
};

export const validator = (candidate) => {
  const vName = validate.validateName(candidate.name);
  const vBalance = validate.validateNumber(candidate.balance);
  const vAgelikeNumber = validate.validateNumber(candidate.age);
  const VAgeLength = validate.validateLength(candidate.age, 3);
  const vEnglish = validate.validateEnglish(candidate.english);
  if (vName || !vBalance || !vAgelikeNumber || VAgeLength || !vEnglish) {
    return false;
  }
  return true;
};

export const cleaner = (nodes) => {
  for (let key in nodes) {
    if (nodes[key].checked === true) {
      nodes[key].checked = false;
      continue;
    }
    nodes[key].value = "";
  }
};

export const createNotif = (str) => {
  // create notif Div
  const notif = document.createElement("div");
  const notifTextDiv = document.createElement("div");
  const notifButtonDiv = document.createElement("div");
  const notifDeleteBtn = document.createElement("button");
  const notifMessage = document.createElement("p");

  // add classes
  notif.classList.add("notification");
  notifButtonDiv.classList.add("notification__button");
  notifTextDiv.classList.add("notification__message");
  notifDeleteBtn.classList.add("button__item");
  notifDeleteBtn.classList.add("button");
  notifMessage.classList.add("message__item");

  //add textContent
  notifDeleteBtn.textContent = "X";
  notifMessage.textContent = str;

  //append chields
  notifTextDiv.appendChild(notifMessage);
  notifButtonDiv.appendChild(notifDeleteBtn);
  notif.appendChild(notifButtonDiv);
  notif.appendChild(notifTextDiv);
  document.body.appendChild(notif);

  // add event Listener
  notifDeleteBtn.addEventListener("click", notifKill.bind(null, notif));
};

export const mutateObject = (obj) => {
  obj.balance = Number(obj.balance);
  obj.age = Number(obj.age);
  return obj;
};

export const drow = (candidates, canvas, context) => {
  var box_Height = canvas.height;
  context.fillStyle = "rgba(94, 144, 252, 1)";
  context.fillRect(1, 1, canvas.width - 1, box_Height - 1);
  context.strokeRect(0.5, 0.5, canvas.width - 1, box_Height);
  const centerX = 100;
  const radius = 40;
  const distance_Between = 105;
  for (let i = 0; i < candidates.length; i++) {
    context.beginPath();
    context.font = " 12px Arial";
    context.fillStyle = "white";
    context.fillText(
      candidates[i].name,
      92 - candidates[i].name.length * 2,
      distance_Between * (i + 1) + 2
    );
    context.closePath();
    context.beginPath();
    context.font = " 12px Arial";
    context.fillStyle = "white";
    context.fillText(
      `check balance`,
      300 * 1.5,
      distance_Between * (i + 1) + 2
    );
    context.closePath();
    context.beginPath();
    context.font = " 12px Arial";
    context.fillStyle = "white";
    context.fillText(`check age`, 500 * 1.5, distance_Between * (i + 1) + 2);
    context.closePath();
    context.beginPath();
    context.font = " 12px Arial";
    context.fillStyle = "white";
    context.fillText(
      `check documents`,
      700 * 1.5,
      distance_Between * (i + 1) + 2
    );
    context.closePath();
    context.beginPath();
    context.font = " 12px Arial";
    context.fillStyle = "white";
    context.fillText(
      `check english`,
      900 * 1.5,
      distance_Between * (i + 1) + 2
    );
    context.closePath();
    let x = 0;
    let k = (2 * Math.PI) / 100;
    const kek = distance_Between * (i + 1) + 2;
    console.log(kek);
    drowCircle(context, canvas, x, k, kek);
  }
};

export function drowCircle(context, canvas, x, k, kek) {
  context.beginPath();
  context.arc(100, kek, 50, 0, x * k, false);
  context.lineWidth = 3;
  context.lineCap = "butt";
  context.strokeStyle = "green";
  context.stroke();
  context.closePath();
  let timeout;
  if (x <= 100) {
    timeout = setTimeout(() => {
      drowCircle(context, canvas, x + 1, k, kek);
    }, 50);
  } else clearTimeout(timeout);
}
