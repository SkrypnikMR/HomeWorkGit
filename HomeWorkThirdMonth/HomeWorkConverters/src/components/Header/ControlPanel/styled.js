import styled from 'styled-components';

export const StControl = styled.div`
background-color: #171730;
width:70%;
color: white;
min-height: 13vh;
display: flex;
justify-content: center;
align-items: center;
    h1{
        font-size: 20px;
        margin: 10px;
    }
`;
export const StModal = {
    content: {
        top: '50%',
        left: '50%',
        right: 'auto',
        bottom: 'auto',
        marginRight: '-50%',
        transform: 'translate(-50%, -50%)',
        width: '40%',
        height: '80vh',
        borderRadius: '5px',
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'flex-end',
        justifyContent: 'flex-start',

        background: 'rgba(0,0,0,0.6)',
    },
};
export const StNotes = styled.div`
    display:flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
    width: 90%;
    margin: 10px;
    color: white;
`;
export const StUl = styled.ul`
    text-decoration: none;
`;
export const StLi = styled.li`
    display: flex;
    font-size: 25px;
    justify-content: space-evenly;
    width: 100%;
        span{
            margin: 10px;
            color:yellow;
        }
        p{
            color:blue;
        }
`;
