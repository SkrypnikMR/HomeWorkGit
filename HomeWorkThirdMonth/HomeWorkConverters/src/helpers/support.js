export const support = {
    lsGet: name => localStorage.getItem(name),
    lsSet: (name, item) => {
        if (typeof item === 'string') return localStorage.setItem(name, item);
        localStorage.setItem(name, JSON.stringify(item));
    },
    convert: (num, primary, secondary) => {
        const lengthSystem = {
            meters: {
                meters: 1,
                miles: 1609.344,
                yards: 1.093613,
                feet: 3.28084,
                versts: 1066.8,
            },
            miles: {
                meters: 1609.344,
                miles: 1,
                yards: 1760,
                feet: 0.003281,
                versts: 1.50857,
            },
            yards: {
                meters: 0.9144,
                miles: 0.000568182,
                yards: 1,
                feet: 3,
                versts: 0.000857143,
            },
            feet: {
                meters: 0.3048,
                miles: 0.000189394,
                yards: 0.333333,
                feet: 1,
                versts: 0.000285714,
            },
            versts: {
                meters: 1066.8,
                miles: 0.662879,
                yards: 1166.67,
                feet: 3500,
                versts: 1,
            },
        };
        return (+num * lengthSystem[primary][secondary]).toString();
    },
};
