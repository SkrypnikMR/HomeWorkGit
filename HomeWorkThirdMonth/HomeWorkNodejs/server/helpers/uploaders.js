require('dotenv/config');
const uuid = require('uuid/v4');
const AWS = require('aws-sdk');
const sharp = require('sharp');
const multer = require('multer');
const s3 = new AWS.S3({
    accessKeyId: process.env.AWS_ID,
    secretAccessKey: process.env.AWS_SECRET,
})

const fileUploader = async (req, res) => {
    const myFile = req.file.originalname.split('.');
    const fileType = myFile[myFile.length - 1];
    try {
        const params = {
            Bucket: process.env.AWS_BUCKET_NAME_FILES,
            Key: `file-${myFile[0]}-${uuid()}.${fileType}`,
            Body: req.file.buffer
        }
        const result = await s3.upload(params).promise().catch((e) => res.status(500).json({ message: 'server error' }));
        return res.status(200).json({ fileLink: result.Location });
    } catch (e) { res.status(500).json({ message: 'server error' }) }
}

const imageUploader = async (req, size, name) => {
    try {
        const myFile = req.file.originalname.split('.');
        const fileType = myFile[myFile.length - 1];

        const sizedImage = await sharp(req.file.buffer).resize({ width: size, height: size }).toBuffer();
        const params = { Bucket: process.env.AWS_BUCKET_NAME_IMAGES, Key: `${name}${myFile[0]}-${uuid()}.${fileType}`, Body: sizedImage }
        const result = await s3.upload(params).promise();

        return result.Location;
    } catch (e) {
        res.status(500).json({ message: 'server error' })
    }

}
const storage = multer.memoryStorage({ destination: (req, res, cb) => cb(null, '') });
const upload = multer({ storage, limits: { fileSize: 20971520 } }).single('file');

const corsOptions = {
    methods: ['POST', 'OPTIONS', 'GET'],
    origins: '*',
    allowedHeaders: ['Content-Type', 'multipart/form-data']
}

module.exports = { fileUploader, imageUploader, upload}