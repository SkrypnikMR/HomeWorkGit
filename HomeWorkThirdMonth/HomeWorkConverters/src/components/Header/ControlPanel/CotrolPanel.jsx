import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { StControl } from './styled';
import Button from '../../UI/Button';

class ControlPanel extends Component {
    handleBtnOnclick = (e) => {
        const { setConverterType, converterType, setConverterTo, setConverterFrom } = this.props;
        setConverterType({ value: e.target.id });
        if (converterType !== 'lengths') {
            setConverterFrom('meters');
            setConverterTo('versts');
        } else {
            setConverterFrom('UAH');
            setConverterTo('EUR');
        }
    }

    render() {
        const { converterType } = this.props;
        return (
            <StControl >
                <h1>Change converter to</h1>
                <Button
                    id="lengths"
                    name="converter_Lengths"
                    borderRadius="0px"
                    height="50px"
                    width="100px"
                    margin="0 10px 0 0"
                    content="Length"
                    disabled={converterType === 'lengths'}
                    onClick={this.handleBtnOnclick}
                />
                <h1>Or</h1>
                <Button
                    id="curency"
                    name="converter_Currency"
                    borderRadius="0px"
                    height="50px"
                    width="100px"
                    margin="0 10px 0 0"
                    disabled={converterType !== 'lengths'}
                    content="Currency"
                    onClick={this.handleBtnOnclick}
                />
            </StControl >
        );
    }
}
ControlPanel.propTypes = {
    setConverterType: PropTypes.func.isRequired,
    converterType: PropTypes.string.isRequired,
    setConverterFrom: PropTypes.func.isRequired,
    setConverterTo: PropTypes.func.isRequired,
};

export default ControlPanel;
