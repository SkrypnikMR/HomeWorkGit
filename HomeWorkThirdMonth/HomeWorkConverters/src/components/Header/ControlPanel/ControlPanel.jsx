import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Modal from 'react-modal';
import { StControl, StModal, StNotes, StUl, StLi } from './styled';
import Button from '../../UI/Button';
import { support } from '../../../helpers/support';

class ControlPanel extends Component {
    constructor(props) {
        super(props);
        this.state = {
            showModal: false,
        };
    }

    handleOpenModal() {
        this.setState({ showModal: true });
    }

    handleCloseModal() {
        this.setState({ showModal: false });
    }

    handleBtnOnclick = (e) => {
        const { setConverterType, converterType, setConverterTo, setConverterFrom } = this.props;
        setConverterType(e.target.id);
        support.lsSet('converterType', e.target.id);
        if (converterType !== 'lengths') {
            setConverterFrom('meters');
            setConverterTo('versts');
        } else {
            setConverterFrom('UAH');
            setConverterTo('EUR');
        }
    }

    render() {
        const { converterType, notes } = this.props;
        const { showModal } = this.state;
        return (
            <StControl >
                <h1>Change converter to</h1>
                <Button
                    id="lengths"
                    name="converter_Lengths"
                    borderRadius="0px"
                    height="50px"
                    width="100px"
                    margin="0 10px 0 0"
                    content="Length"
                    disabled={converterType === 'lengths'}
                    onClick={this.handleBtnOnclick}
                />
                <h1>Or</h1>
                <Button
                    id="curency"
                    name="converter_Currency"
                    borderRadius="0px"
                    height="50px"
                    width="100px"
                    margin="0 10px 0 0"
                    disabled={converterType !== 'lengths'}
                    content="Currency"
                    onClick={this.handleBtnOnclick}
                />
                <Button
                    id="modal_btn"
                    name="modal_btn"
                    borderRadius="0px"
                    height="50px"
                    width="100px"
                    margin="0 10px 0 0"
                    content="Show All Results"
                    onClick={this.handleOpenModal.bind(this)}
                />
                <Modal
                    isOpen={showModal}
                    contentLabel="Minimal Modal Example"
                    ariaHideApp={false}
                    style={StModal}
                >
                    <Button
                        id="modal_btn-close"
                        name="modal_btn-close"
                        borderRadius="0px"
                        height="50px"
                        width="100px"
                        margin="0 10px 0 0"
                        content="X"
                        onClick={this.handleCloseModal.bind(this)}
                    />
                    <StNotes>
                        <h1>Your Converter Results</h1>
                        <StUl>
                            {notes.length > 0 ? notes.map((el) => {
                                return (
                                    <StLi key={`${el.tablo}${el.from}`}>
                                        <p>
                                            {el.tablo}
                                            <span>{el.from.toUpperCase()}</span>
                                            from
                                            <span>{el.to.toUpperCase()}</span>
                                        </p>
                                    </StLi>
                                );
                            }) : <p>You don&apos;t have some notes</p>}
                        </StUl>
                    </StNotes>
                </Modal>
            </StControl >
        );
    }
}
ControlPanel.propTypes = {
    setConverterType: PropTypes.func.isRequired,
    converterType: PropTypes.string.isRequired,
    setConverterFrom: PropTypes.func.isRequired,
    setConverterTo: PropTypes.func.isRequired,
    notes: PropTypes.arrayOf(PropTypes.objectOf(PropTypes.string)),
};

export default ControlPanel;
