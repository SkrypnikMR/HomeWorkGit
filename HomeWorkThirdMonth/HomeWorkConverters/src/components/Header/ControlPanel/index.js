import { connect } from 'react-redux';
import ControlPanel from './CotrolPanel.jsx';
import { setConverterType, setConverterFrom, setConverterTo } from '../../../store/converter/actions';

const MapStateToProps = state => (state);

const mapDispatchToProps = dispatch => ({
    setConverterType: payload => dispatch(setConverterType(payload)),
    setConverterFrom: payload => dispatch(setConverterFrom(payload)),
    setConverterTo: payload => dispatch(setConverterTo(payload)),
});

export default connect(MapStateToProps, mapDispatchToProps)(ControlPanel);
