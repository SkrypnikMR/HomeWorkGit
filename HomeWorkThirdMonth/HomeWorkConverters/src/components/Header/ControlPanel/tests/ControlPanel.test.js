import React from 'react';
import configureStore from 'redux-mock-store';
import { shallow } from 'enzyme';
import { shallowSmart, mountSmart } from '../../../../../__tests__/helpers/testHelper';
import ControlPanel from '../ControlPanel';

jest.mock('react-modal');

const mockStore = configureStore();
const store = mockStore({
    isLoading: false,
    converterType: 'lengths',
    from: '',
    to: '',
    quantity: '',
    tablo: '',
    currency: {},
    notes: [],
});

describe('Button', () => {
    it('Should match snapshot', () => {
        const component = shallowSmart(<ControlPanel notes={[{
            tablo: 'kek',
            from: 'shrek',
            to: 'fiona',
        }]}
        />, store);
        expect(component.html()).toMatchSnapshot();
    });
    it('should render Header', () => {
        const component = mountSmart(<ControlPanel notes={[]} />, store);
        expect(component.find('Button')).toHaveLength(3);
    });
    it('should render Converter', () => {
        const component = mountSmart(<ControlPanel notes={[]} />, store);
        expect(component.find('h1')).toHaveLength(2);
    });
    it('should click on lengths button', () => {
        const setConverterType = jest.fn();
        const converterType = 'currency';
        const setConverterTo = jest.fn();
        const setConverterFrom = jest.fn();

        const component = mountSmart(
            <ControlPanel
                setConverterType={setConverterType}
                converterType={converterType}
                setConverterTo={setConverterTo}
                setConverterFrom={setConverterFrom}
                notes={[]}
            />, store);
        component.find('Button').first().simulate('click');
        expect(setConverterType).toHaveBeenCalled();
        expect(setConverterFrom).toHaveBeenCalled();
        expect(setConverterTo).toHaveBeenCalled();
    });
    it('should click on currency button', () => {
        const setConverterType = jest.fn();
        const converterType = 'lengths';
        const setConverterTo = jest.fn();
        const setConverterFrom = jest.fn();

        const component = mountSmart(
            <ControlPanel
                setConverterType={setConverterType}
                converterType={converterType}
                setConverterTo={setConverterTo}
                setConverterFrom={setConverterFrom}
                notes={[]}
            />, store);
        component.find('Button').at(1).simulate('click');
        expect(setConverterType).toHaveBeenCalled();
        expect(setConverterFrom).toHaveBeenCalled();
        expect(setConverterTo).toHaveBeenCalled();
    });
    it('should click on modal button', () => {
        const component = shallow(<ControlPanel notes={[]} />);
        component.find('Button').at(2).simulate('click');
        expect(component.state().showModal).toBe(true);
    });
    it('should click on close button', () => {
        const component = shallow(<ControlPanel notes={[]} />);
        component.find('Button').at(2).simulate('click');
        component.find('Button').at(3).simulate('click');
        expect(component.state().showModal).toBe(false);
    });
    it('should render li with notes ', () => {
        const component = shallow(<ControlPanel notes={[{
            tablo: 'kek',
            from: 'shrek',
            to: 'fiona',
        }]}
        />);
        expect(component.find('styled__StLi')).toHaveLength(1);
    });
});
