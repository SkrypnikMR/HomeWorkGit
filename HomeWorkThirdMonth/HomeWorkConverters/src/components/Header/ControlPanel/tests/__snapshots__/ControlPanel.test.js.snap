// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should match snapshot 1`] = `"<div class=\\"styled__StControl-sc-13jhp11-0 hMnvOa\\"><h1>Change converter to</h1><button id=\\"lengths\\" name=\\"converter_Lengths\\" type=\\"button\\" height=\\"50px\\" width=\\"100px\\" class=\\"styled__StButton-sc-1a6hck8-0 kBWHlo\\">Length</button><h1>Or</h1><button id=\\"curency\\" name=\\"converter_Currency\\" type=\\"button\\" disabled=\\"\\" height=\\"50px\\" width=\\"100px\\" class=\\"styled__StButton-sc-1a6hck8-0 kBWHlo\\">Currency</button><button id=\\"modal_btn\\" name=\\"modal_btn\\" type=\\"button\\" height=\\"50px\\" width=\\"100px\\" class=\\"styled__StButton-sc-1a6hck8-0 kBWHlo\\">Show All Results</button></div>"`;
